### PR TITLE
docs(refactor): moves pv recovery and FAQs to separate sections

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Doc refactor and edit: moves the content related to recovering a cluster using PVs and FAQS to separate sections.
Final part of the process to move content out of the _Managing_ chapter.  

- Moves the instructions to recover using PVs into root-level assembly with rewrites
- Moves the FAQs content to appropriate sections with rewrites:
    - Cluster Operator deployed resources requiring admin privileges to Deploying the Cluster Operator  
    - Hostname verification with node ports to Securing access to a Kafka cluster
    - Troubleshooting TLS hostname verification with node ports to Securing access to a Kafka cluster
    - Lock acquisition warnings for cluster operations to Configuring logging levels
    - Need for ClusterRoleBindings to Delegating privileges to Strimzi components

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/configuring/con-failed-lock-warnings.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+[source,shell]
+----
+WARNING AbstractClusterOperations:290 - Failed to acquire lock for kafka cluster lock::kafka::myproject::my-cluster
+----
+
+In configurations where `STRIMZI_FULL_RECONCILIATION_INTERVAL_MS` and `STRIMZI_OPERATION_TIMEOUT_MS` are set, occasional warnings may occur without underlying issues. 
+Timed-out operations are retried during the next periodic reconciliation.
+
+Persistent appearance of this warning, especially when no other operations should be running, might indicate a lock release error. 
+Restarting the Cluster Operator can resolve such issues.
```
Comment (scholzj) [inline documentation/modules/configuring/con-failed-lock-warnings.adoc:26]: This is valid. But given this message is no on DEBUG level, would it be more useful to suggest this for when log messages like this:

```
2024-06-19 12:20:46 INFO  AbstractOperator:399 - Reconciliation #1(watch) Kafka(myproject/my-cluster): Reconciliation is in progress
```

keep appearing for a long time with the same reconciliation number?

**Review by scholzj — COMMENTED** [review]
Some nits to one of the parts - likely due to the original docs being outdated.

**Review by scholzj — APPROVED** [review]

File: documentation/assemblies/deploying/assembly-deploy-cluster-operator.adoc [hunk 2]
```diff
@@ -6,9 +6,22 @@
...
 
-When the Cluster Operator is running, it starts to watch for updates of Kafka resources.
+The command sets up everything you need to be able to create and manage a Kafka deployment, including the following resources:
+
+* Cluster Operator (`Deployment`, `ConfigMap`)
+* Strimzi CRDs (`CustomResourceDefinition`)
+* RBAC resources (`ClusterRole`, `ClusterRoleBinding`, `RoleBinding`)
+* Service account (`ServiceAccount`) 
+
+Cluster-scoped resources like `CRDs`, `ClusterRole`, and `ClusterRoleBinding` require administrator privileges for installation.
```
Comment (ppatierno) [inline documentation/assemblies/deploying/assembly-deploy-cluster-operator.adoc:19]: Shouldn't we use the long `CustomResourceDefinition` name as the others?

**Review by ppatierno — APPROVED** [review]
LGTM. Just one nit.

**Pull request merged**